server/api_device_stats.go

```diff
@@ -40,13 +40,34 @@ type DeviceStats struct {
 	ImpactRates  [2016]float64
 }
 
+//  DeviceStatsFriendlyJSON is strictly for JSON encoding
+// It is a friendly version of DeviceStats that includes the PublicKey as a hex string
+// and also includes the ShortId of the device to avoid having to look it up in another HTTP request
+// It contains pointers to the arrays to avoid copying the data
+// DeviceStatsFriendlyJSON's lifetime is limited to the HTTP request by design
+type DeviceStatsFriendlyJSON struct {
+	PublicKey    string
+	ShortId 	 *uint32
+	PowerOutputs *[2016]uint64
+	ImpactRates  *[2016]float64
+}
+
 // AllDeviceStats contains aggregate weekly statistics
 type AllDeviceStats struct {
 	Devices        []DeviceStats // A set of data for each device
 	TimeslotOffset uint32        // Establish what week is covered by the data
 	Signature      glow.Signature
 }
 
+// AllDeviceStatsFriendlyJSON is strictly for JSON encoding
+// It is a friendly version of AllDeviceStats that includes the PublicKey as a hex string
+// and also includes the ShortId of the device to avoid having to look it up in another HTTP request
+type AllDeviceStatsFriendlyJSON struct {
+	Devices        []DeviceStatsFriendlyJSON // A set of data for each device
+	TimeslotOffset uint32        // Establish what week is covered by the data
+	Signature      glow.Signature
```

Comment:
You need to update the signature code on this to cover the new fields. The API isn't protected by https, so you have to assume MiTM attacks. If you aren't verifying the signatures, you could be reading corrupted data.